File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/data/api/services/ApiMovieService.java
package com.crpr.androidcinema.data.api.services;

import com.crpr.androidcinema.data.api.models.ApiMovie;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by claudioribeiro on 31/01/2017.
 */
public class ApiMovieService {

    public interface ServiceClient {
        @GET("movie/{movie_id}")
        Observable<ApiMovie> getMovieDetailsById(@Query("movie_id") int movieId);
    }

    private final ApiMovieService.ServiceClient _client;

    public ApiMovieService(ApiMovieService.ServiceClient client){
        this._client = client;
    }

    public Observable<ApiMovie> getMovieDetailsById(int token) {
        return _client.getMovieDetailsById(token);
    }

}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/data/api/services/ApiMovieService.kt
package com.crpr.androidcinema.data.api.services

import com.crpr.androidcinema.data.api.models.ApiMovie

import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

/**
 * Created by claudioribeiro on 31/01/2017.
 */
class ApiMovieService(private val _client: ApiMovieService.ServiceClient) {

    interface ServiceClient {
        @GET("movie/{movie_id}")
        fun getMovieDetailsById(@Query("movie_id") movieId: Int): Observable<ApiMovie>
    }

    fun getMovieDetailsById(token: Int): Observable<ApiMovie> {
        return _client.getMovieDetailsById(token)
    }

}
